### PR TITLE
Issue/integrate bedita plugins

### PR DESCRIPTION
**Pull request by vallo87**
introduced yarn/npm scripts (webpack.config.plugin.js):
- build-plugins: bundles every plugins under plugins folder
- dev-plugins: start watching every plugins under plugins folder for dev

Main app:
load BEplugins and dynamically load and register Vue components

webpack.config refactor, moved common logic in one file

### CODE REVIEW

File: src/Template/Element/json_meta_config.ctp [hunk 1]
```diff
@@ -12,7 +12,21 @@
...
         'action': '{$view->action|default:"index"}',
-        'relations': {}
+        'relations': {},
+        'plugins': '<?= json_encode(\Cake\Core\Configure::read('Plugins')) ?>'
     };
 
+    if (BEDITA.plugins) {
+        try {
+            BEDITA.plugins = JSON.parse(BEDITA.plugins);
+            if (BEDITA.plugins.hasOwnProperty('DebugKit')) {
```
Comment (stefanorosanelli) [inline src/Template/Element/json_meta_config.ctp]: There's no harm if `DebugKit` is here... we can leave it in `BEDITA.plugins`

File: webpack.config.js [hunk 2]
```diff
@@ -1,18 +1,8 @@
...
-        console.log(chalk.red.bold(separator));
-        console.log(chalk.blue.bold(message));
-        console.log(chalk.red.bold(separator));
-    }
-}
-
-// node dependencies
-const path = require('path');
-const chalk = require('chalk');
+// erquire environment setup
```
Comment (stefanorosanelli) [inline webpack.config.js]: typo `require`

File: webpack.config.plugin.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
+// erquire environment setup
+require('./webpack.config.environment');
+
+const webpack = require('webpack');
+// const ExtractTextPlugin = require('extract-text-webpack-plugin');
```
Comment (stefanorosanelli) [inline webpack.config.plugin.js]: are these lines necessary?

File: webpack.config.plugin.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+                    presets: [
+                        ['@babel/preset-env', {
+                            modules: false,
+                            browsers: ['> 99%'],
+                            useBuiltIns: "usage",
+                        }]
+                    ]
+                }
+            },
+            // {
```
Comment (stefanorosanelli) [inline webpack.config.plugin.js]: are these commented lines necessary?

File: webpack.config.js (hunk 2 above)
Comment (vallo87) [reply webpack.config.js]: my bad!

File: webpack.config.plugin.js (hunk 3 above)
Comment (vallo87) [reply webpack.config.plugin.js]: not necessary for now, it was a work in progress for scss parsing

File: webpack.config.plugin.js (hunk 4 above)
Comment (vallo87) [reply webpack.config.plugin.js]: same as the above comment

File: src/Template/Layout/default.twig [hunk 5]
```diff
@@ -84,6 +84,11 @@ BEdita 4 ~ @ 2018 from ChannelWeb & Chialab with love
 
     {{ Html.script('manifest.bundle')|raw }}
     {{ Html.script('vendors.bundle')|raw }}
+
+    {{ Html.script('vendors.bundle')|raw }}
```
Comment (didoda) [inline src/Template/Layout/default.twig]: duplicate row? it's like line 86

**Pull request merged**